platform/ios/src/MGLMapView.mm

```diff
@@ -2426,9 +2450,12 @@ - (void)addAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)annotations
         
         for (size_t i = 0; i < pointAnnotationTags.size(); ++i)
         {
+            MGLAnnotationImage *annotationImage = annotationImages[i];
+            annotationImage.styleIconIdentifier = @(points[i].icon.c_str());
+            
             MGLAnnotationContext context;
             context.annotation = annotations[i];
-            context.symbolIdentifier = @(points[i].icon.c_str());
+            context.imageReuseIdentifier = annotationImage.reuseIdentifier;
```

Comment:
Great! I made a similar refactor on my annotation branch. This is better and includes the new and better property name on MGLAnnotationContext